### PR TITLE
12 3.19.4 bits

### PR DESCRIPTION
**Pull request by mingwandroid**

### CODE REVIEW

**Comment by mingwandroid** [comment]
@jjhelmus preview, I will also check some community PRs once you (and ci) are happy with all this!

File: conda_build/build.py [hunk 1]
```diff
@@ -973,12 +998,13 @@ def record_prefix_files(m, files_with_prefix):
 
         print("Files containing CONDA_PREFIX")
         print("-----------------------------")
-        detect_binary_files_with_prefix = m.get_value('build/detect_binary_files_with_prefix', False)
+        detect_binary_files_with_prefix = m.get_value('build/detect_binary_files_with_prefix',
+                                                      not len_binary_has_prefix_files and not utils.on_win)
```
Comment (jjhelmus) [inline conda_build/build.py:1002]:
```suggestion
                                                      len_binary_has_prefix_files and not utils.on_win)
```
The default should be True when there are binary prefix files and the platform is not windows

Comment (mingwandroid) [reply conda_build/build.py:1002]: > when there are binary prefix files

This is not the thing being checked here. `not len_binary_has_prefix_files` means the user hasn't specified any binary files, so we need to check them all (on non Windows).

The goal is to be able to control which binaries get prefix replaced and which do not. I think this change is consistent with the documentation: https://docs.conda.io/projects/conda-build/en/latest/resources/make-relocatable.html#making-packages-relocatable

> Any binary file containing the build prefix can automatically be registered in info/has_prefix using build/detect_binary_files_with_prefix in meta.yaml. Alternatively, individual binary files can be registered by listing them in build/binary_has_prefix_files in meta.yaml. The registered files will have their build prefix replaced with the install prefix at install time. This works by padding the install prefix with null terminators, such that the length of the binary file remains the same. The build prefix must therefore be long enough to accommodate any reasonable installation prefix. On macOS and Linux, conda-build pads the build prefix to 255 characters by appending _placeholds to the end of the build directory name.

I will look at any test failures soon.

Comment (jjhelmus) [reply conda_build/build.py:1002]: Yes, you are completely correct.  I mis-understood the `len_binary_has_prefix_files` variable.

**Comment by mingwandroid** [comment]
I think this is good to go now. If you have time and agree, I'll look over community PRs after some food.

**Pull request merged**